modules/wallet/encrypt_test.go

```diff
@@ -284,3 +284,32 @@ func TestInitFromSeed(t *testing.T) {
 		t.Fatalf("wallet should have correct balance after loading seed: wanted %v, got %v", origBal, newBal)
 	}
 }
+
+// TestReencrypt tests that Reencrypt re-encrypts a wallet correctly.
+func TestReencrypt(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	wt, err := createBlankWalletTester(t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer wt.closeWt()
+
+	var masterKey crypto.TwofishKey
+	fastrand.Read(masterKey[:])
+	_, err = wt.wallet.Encrypt(masterKey)
+	if err != nil {
+		t.Error(err)
+	}
+
+	var encryptionKey crypto.TwofishKey
+	seed, err := wt.wallet.Reencrypt(encryptionKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	walletKey := crypto.TwofishKey(crypto.HashObject(seed))
+	postEncryptionTesting(wt.miner, wt.wallet, walletKey)
+}
```

Comment:
oh, I see. Ok, so I think for this PR it's sufficient to:

+ add a call to postEncryptionTesting after encrypting the wallet the first time
+ add a second send to postEncryptionTesting that drains the wallet entirely. You may need to change the first send if it's already draining the wallet

Ideally we'd find a way to slip in receiving coins as well, but I think that the way it's coded up, just the change address from the first send (as long as it's not the whole wallet) will be enough to get the same code coverage. But that's implementation-specific coverage instead of blackbox testing coverage.